src/masternodes/mn_checks.cpp

```diff
@@ -2171,6 +2180,164 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
         return Res::Ok();
     }
 
+    
+    Res operator()(const CDUSDLockMessage& obj) const {
+        auto res = CheckCustomTx();
+        if (!res)
+            return res;
+
+        if (!HasAuth(obj.owner)) {
+            return Res::Err("Transaction must have at least one input from owner");
+        }
+
+        const auto attributes = mnview.GetAttributes();
+        if (!attributes) {
+            return Res::Err("Attributes unavailable");
+        }
+
+        const auto paramID = ParamIDs::DFIP2211D;
+
+        CDataStructureV0 activeKey{AttributeTypes::Param, paramID, DFIPKeys::Active};
+        CDataStructureV0 limitKey{AttributeTypes::Param, paramID, DFIPKeys::Limit, obj.batchId};
+        CDataStructureV0 tokenKey{AttributeTypes::Param, paramID, DFIPKeys::LockToken, obj.batchId};
+        if (!attributes->GetValue(activeKey, false)){
+            return Res::Err("DFIP2211D not currently active");
+
+        }
+        if(!attributes->CheckKey(limitKey) ||
+            !attributes->CheckKey(tokenKey)) {
+            return Res::Err("This batch is currently not available");
+        }
+
+        auto dusdToken = mnview.GetToken("DUSD");
+        if(!dusdToken) {
+            return Res::Err("Could not find dusd token");
+        }
+
+
+        auto tokenId= attributes->GetValue(tokenKey, (uint32_t)0);
+        auto limit = attributes->GetValue(limitKey,(uint32_t)0)*COIN;
+        if(tokenId == 0) {
+            return Res::Err("missing lock token");
+        }
+        auto lockTokenId = DCT_ID{tokenId};
+        auto lockToken = mnview.GetToken(lockTokenId);
+        if(!lockToken) {
+            return Res::Err("Could not find lock token");
+        }
+        
+        auto pair = mnview.GetPoolPair(lockTokenId, dusdToken->first);
+        if (!pair) {
+            return Res::Err("lock pool not found");
+        }
+        auto& pool = pair->second;
+        const auto& lpTokenID = pair->first;
+        
+        if(obj.source.nTokenId != dusdToken->first && 
+            obj.source.nTokenId != lockTokenId && 
+            obj.source.nTokenId != lpTokenID) {
+            return Res::Err("Invalid token for this batch. Need to send DUSD to deposit, or the lockToken or LP-Token to withdraw.");
+        }
+
+        CalculateOwnerRewards(obj.owner);
+
+
+        if (obj.source.nTokenId == dusdToken->first) {
+            // case of locking DUSD: "swap" half to locktoken, add to pool and send LP token to user
+
+            res = mnview.SubBalance(obj.owner, obj.source);
+            if (!res)
+                return res;
+
+            auto dusdhalf = obj.source.nValue / 2;
+            auto lockedhalf = obj.source.nValue - dusdhalf;
+
+            if (dusdhalf == 0 || lockedhalf == 0) {
+                return Res::Err("amount too small.");
+            }
+            auto resultedMint = SafeAdd(lockToken->minted, lockedhalf);
```

Comment:
This is redundant, as it is done in `AddMintedTokens` so just remove it.

Reply:
I need resultedMint to check against the limit beforehand not only to check if the add is fine.